app/scripts/migrations/140.ts

```diff
@@ -0,0 +1,251 @@
+ */
+function transformState(state: Record<string, unknown>) {
+  const {
+    NetworkController: networkControllerState,
+    PermissionController: permissionControllerState,
+    SelectedNetworkController: selectedNetworkControllerState,
+  } = state;
+  if (!networkControllerState || !isObject(networkControllerState)) {
+    log.warn(
+      'Skipping migration: `NetworkController` state not found or is not an object.',
+    );
+    return state;
+  }
+
+  if (
+    !hasProperty(networkControllerState, 'selectedNetworkClientId') ||
+    typeof networkControllerState.selectedNetworkClientId !== 'string'
+  ) {
+    log.warn(
+      'Skipping migration: `NetworkController.selectedNetworkClientId` is not a string.',
+    );
+    return state;
+  }
+
+  if (
+    !hasProperty(networkControllerState, 'networkConfigurationsByChainId') ||
+    !isObject(networkControllerState.networkConfigurationsByChainId)
+  ) {
+    log.warn(
+      'Skipping migration: `NetworkController.networkConfigurationsByChainId` is not an object.',
+    );
+    return state;
+  }
+
+  if (!permissionControllerState || !isObject(permissionControllerState)) {
+    log.warn(
+      'Skipping migration: `PermissionController` state not found or is not an object.',
+    );
+    return state;
+  }
+
+  if (!isObject(permissionControllerState.subjects)) {
+    log.warn(
+      'Skipping migration: `PermissionController.subjects` state is not an object.',
+    );
+    return state;
+  }
+
+  if (
+    !selectedNetworkControllerState ||
+    !isObject(selectedNetworkControllerState)
+  ) {
+    log.warn(
+      'Skipping migration: `SelectedNetworkController` state not found or is not an object.',
+    );
+    return state;
+  }
+
+  if (!isObject(selectedNetworkControllerState.domains)) {
+    log.warn(
+      'Skipping migration: `SelectedNetworkController.domains` state is not an object.',
+    );
+    return state;
+  }
+
+  const currentChainId = getChainId(
+    networkControllerState.selectedNetworkClientId,
+    networkControllerState.networkConfigurationsByChainId,
+  );
+  if (typeof currentChainId !== 'string') {
+    log.warn(
+      'Skipping migration: `NetworkController.selectedNetworkClientId` is not a string.',
+    );
+    return state;
+  }
+
+  const updatedSubjects: string[] = [];
+
+  // Add permission to use the current globally selected network to all Snaps
+  // that have the `endowment:ethereum-provider` permission.
+  const entries = Object.entries(permissionControllerState.subjects) as [
+    string,
+    GenericPermissionControllerSubject,
+  ][];
+
+  permissionControllerState.subjects = entries.reduce<
+    PermissionControllerSubjects<PermissionConstraint>
+  >((accumulator, [key, subject]) => {
+    const permissionKeys = Object.keys(subject.permissions);
+    const needsMigration =
+      permissionKeys.includes(SnapEndowments.EthereumProvider) &&
+      !permissionKeys.includes(PermissionNames.permittedChains);
```

Comment:
This looks to be out of date since [this PR was merged yesterday](https://github.com/MetaMask/metamask-extension/pull/27847). `PermissionNames.permittedChains` no longer exists. Instead for this check we will need to examine whether it has a `Caip25EndowmentPermissionName`.

Reply:
Good catch, I forgot to update the permission name here.